### PR TITLE
update actions/checkout in GitHub Actions workflows to v3

### PR DESCRIPTION
**Pull request by striezel**
Updates the `actions/checkout` action used in the GitHub Actions workflows to its newest version.

Changes in [actions/checkout](https://github.com/actions/checkout):

> ## v3.0.2
> - Add input `set-safe-directory`
>
> ## v3.0.1
> - Fixed an issue where checkout failed to run in container jobs due to the new git setting `safe.directory`
> - Bumped various npm package versions
>
> ## v3.0.0
>
> - Update to node 16

As far as I can tell this should all be backwards compatible, so I do not expect any breakage.

### CODE REVIEW

**Review by ptondereau — APPROVED** [review]

**Comment by striezel** [comment]
FYI, the linting errors are fixed in PR #152.

If you wish so, I can rebase this PR after #152 gets merged. That should take care of the failing checks here.

**Comment by TobiasBengtsson** [comment]
@striezel Sorry for the delay, https://github.com/davidcole1340/ext-php-rs/pull/152 is now merged so you can rebase on top of latest `master`.

**Comment by striezel** [comment]
@TobiasBengtsson:
> EDIT: looks like we have new failures instead, ideally I'll have time to fix them toda

I made another pull request for those warnings: https://github.com/davidcole1340/ext-php-rs/pull/157

Not sure what the actual reason for those new warnings is, but a [new version of clippy](https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#rust-164) (and Rust) was released after #152 was created, so maybe that could be it?

**Comment by ptondereau** [comment]
BTW, I think we should only run clippy on stable version

**Comment by TobiasBengtsson** [comment]
Yes, we are using the nightly and it breaks way too often. I agree it would be better to let clippy run on stable

**Comment by striezel** [comment]
> @striezel Sorry for the delay, #152 is now merged so you can rebase on top of latest `master`.

Don't worry, I had worse. :D Less than a month is still OK.
_(My current record PR will hit three years this weekend without a reply from the maintainer - who is still active on GitHub, but does not seem to bother about PRs.)_

I just rebased this PR on top of the current `master` branch. Those checks should pass now.

**Comment by striezel** [comment]
OK, failed with a different error this time, and it's not clippy:
```
error: Unknown `#[clap(long)]` attribute (`#[arg(long)] exists)
  --> crates/cli/src/lib.rs:92:12
   |
92 |     #[clap(long)]
   |            ^^^^

error: Unknown `#[clap(long)]` attribute (`#[arg(long)] exists)
   --> crates/cli/src/lib.rs:115:12
    |
115 |     #[clap(long)]
    |            ^^^^

error: Unknown `#[clap(short)]` attribute (`#[arg(short)] exists)
   --> crates/cli/src/lib.rs:134:12
    |
134 |     #[clap(short, long)]
    |            ^^^^^

error: could not compile `cargo-php` due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
Error: Process completed with exit code 101.
```

Unfortunately, I cannot reproduce this on my local machine, so I am not sure that I can provide a fix for that, too.

**Comment by striezel** [comment]
OK, scratch that last comment about not providing a fix.

I attempted to fix those errors in <https://github.com/davidcole1340/ext-php-rs/pull/158> and so far it looks good, builds are passing.
(Spoiler alert: A new major release of [`clap`, version 4.0.0](https://crates.io/crates/clap/4.0.0), was released a few hours ago earlier today, and that caused the build to break.)

So I will rebase this PR gain, if #158 is merged.

**Comment by striezel** [comment]
> So I will rebase this PR gain, if #158 is merged.

Rebased.

**Pull request merged**
